src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -4249,6 +4265,23 @@ PlacementExecutionDone(TaskPlacementExecution *placementExecution, bool succeede
 	{
 		/* mark the placement execution as finished */
 		placementExecution->executionState = PLACEMENT_EXECUTION_FINISHED;
+
+		Assert(INSTR_TIME_IS_ZERO(placementExecution->endTime));
+		INSTR_TIME_SET_CURRENT(placementExecution->endTime);
+
+		if (IsLoggableLevel(DEBUG4))
+		{
+			long durationMillisecs =
+				MillisecondsBetweenTimestamps(placementExecution->startTime,
+											  placementExecution->endTime);
+
+			ereport(DEBUG4, (errmsg("Task execution (%d) for placement (%ld) finished in "
```

Comment:
could be kind of nice to have the anchor shard id in here

super minor: errmsg should start with lower case